gflownet/envs/crystals/spacegroup.py

```diff
@@ -656,11 +673,92 @@ def build_n_atoms_compatibility_dict(n_atoms: List[int], space_groups: List[int]
         assert all([sg > 0 and sg <= 230 for sg in space_groups])
         return {sg: Group(sg).check_compatible(n_atoms)[0] for sg in space_groups}
 
+    def _restrict_space_groups(self, sg_subset: Optional[Iterable] = None):
+        """
+        Updates the dictionaries:
+            - self.space_groups
+            - self.crystal_lattice_systems
+            - self.point_symmetries
+        by eliminating the space groups that are not in the subset sg_subset passed as
+        an argument.
+        """
+        if sg_subset is None:
+            return
+        sg_subset = set(sg_subset)
+
+        # Update self.space_groups
+        self.space_groups = deepcopy(self.space_groups)
+        sg_to_remove = [sg for sg in self.space_groups if sg not in sg_subset]
+        for sg in sg_to_remove:
+            del self.space_groups[sg]
```

Comment:
Is there a reason to do it this way (figure out what to remove and then remove it) rather than a simpler : 
`self.space_groups = [sg for sg in self.space_groups if sg in sg_subset]` ?

Reply:
I don't even know why I did something like that. I guess pre-deadline brain death? Updated in https://github.com/alexhernandezgarcia/gflownet/pull/223/commits/75aa91e788ab5d58f308d26b85a9c266366418a1.

Reply:
Ok, I just figured out why you did it that way... because self.space_groups isn't a list, it's a dictionary. 
Which is why the tests are currently failing; the new version of the code replaces dicts by lists.
Still, you can keep the simpler code and just do something like :  
`{k: v for (k, v) in self.space_groups.items() if k in sg_subset}`
to get a dict instead.

Reply:
Ah right! https://github.com/alexhernandezgarcia/gflownet/pull/223/commits/0a10e03aefc87468583b41dbd4ff48fe0a70f361